support/connection/registry.go

```diff
@@ -0,0 +1,90 @@
+package connection
+
+import (
+	"fmt"
+
+	"github.com/project-flogo/core/support/managed"
+	"github.com/project-flogo/core/support/log"
+)
+
+var (
+	managerFactories = make(map[string]ManagerFactory)
+	managers = make(map[string]Manager)
+)
+
+func RegisterManagerFactory(connectionType string, factory ManagerFactory) error {
+
+	if connectionType == "" {
+		return fmt.Errorf("'connectionType' must be specified when registering")
+	}
+
+	if factory == nil {
+		return fmt.Errorf("cannot register with 'nil' manager factory")
+	}
+
+	if _, dup := managerFactories[connectionType]; dup {
+		return fmt.Errorf("manager factory already registered for type: %s", connectionType)
+	}
+
+	log.RootLogger().Debugf("Registering manager factory for type: %s", connectionType)
+
+	managerFactories[connectionType] = factory
+
+	return nil
+}
+
+func GetManagerFactory(id string) ManagerFactory {
+	return managerFactories[id]
+}
+
+func RegisterManager(connectionId string, manager Manager) error {
+
+	if connectionId == "" {
+		return fmt.Errorf("'id' must be specified when registering")
+	}
+
+	if manager == nil {
+		return fmt.Errorf("cannot register with 'nil' manager")
+	}
+
+	if _, dup := managers[connectionId]; dup {
+		return fmt.Errorf("manager already registered: %s", connectionId)
+	}
+
+	log.RootLogger().Debugf("Registering manager: %s", connectionId)
+
+	managers[connectionId] = manager
+
+	return nil
+}
+
+func GetManager(id string) Manager {
+	return managers[id]
+}
+
+func StartManagers() error {
```

Comment:
@fm-tibco : Rather than engine managing lifecycle for all connection types , should we let ManagerFactory do that? As part of engine start/stop, we call Factory who would handle connection of its own type.

Reply:
I cleaned it up a bit in the latest commit.  Think of this as analogous... a ConnectionPool and a ConnectionPoolFactory... I think the job of the factory is to create the pools... and you would lifecycle the ConnectionPools themselves.  I'll think about it some more... @mellistibco @skothari-tibco ... any thoughts on this?

Reply:
@fm-tibco: Do you think supporting first class **connection** data type would be appropriate now?
e.g. An activity can declare connection data type with following data which engine can use to resolve appropriate connection instance.
```
"httpConnection": {
                                    "type": "http",
                                    "id": "myhttpconnection"
                               }
```

This would simplify contribution code as they don't need to call connection.GetManager(). Rather engine will do that and set appropriate connection instance during initialization.

Reply:
I like the idea... it should either accommodate the id or the settings.. so people don't need special code to handle the different setups

inline:  
```
"httpConnection": 
    {
        "type": "http",
        "settings": {"":""}
    }
```  
by ref/shared:
```
"httpConnection": 
    {
        "type": "http",
        "id": "myhttpconnection"
    }
```

Reply:
I'll update the PR... @mellistibco, let me know what you think when you have a chance

Reply:
@fm-tibco : I dont see new connection type added for the contribution developers. Am I missing something ?

Reply:
+1 I like that also

Reply:
@vijaynalawade are you talking about "connection" data type?  or are you talking about the connection.Manager & connection.Manager interfaces?  I haven't had a chance to check in the connection data type stuff.